### PR TITLE
fmt Cargo.tomls

### PR DESCRIPTION
**Pull request by gilescope**
We autoformat the code so why not the Cargo.tomls?

There's some excellent options:

https://taplo.tamasfe.dev/configuration/#formatting-options

(There's a command line tool to do the formatting as well as a toml language server and a vscode extension called `even better toml`.)

I personally think that the vertical alignment makes them much more readable and the exceptions  stand out more.
```
# Align consecutive entries vertically.
align_entries = true
```

Am temped to try:
```
# Alphabetically reorder keys that are not separated by empty lines.
reorder_keys = true
```
(Will give that a shot tomorrow.)

Curious what others think?

### CODE REVIEW

**Comment by gilescope** [comment]
3rd party dependencies go first seems to be the current convention so will do that.

**Comment by s3krit** [comment]
Looks good. I've only eyeballed the Cargo.tomls but since it's just formatting changes that seems ok. Would be nice to have a CI job that checks the formatting else we'll just have to run this occasionally to ensure the formatting doesn't drift over time. If we had pre-merge hooks that could do that automagically that would be even better but I think we're a ways off that.

**Review by s3krit — APPROVED** [review]

**Comment by athei** [comment]
Looks good to me. I second the request for a CI job.

**Comment by bkchr** [comment]
I'm not sure about introducing another formatting tool. This tool is not even 1 year old, so we don't know what will happen with it when something breaks. I would really not want a breaking CI because of such a tool.

Regarding the vertical formatting, I think this wastes tons of space and it is also not using the same formatting when dependencies are grouped.

File: Cargo.toml [hunk 1]
```diff
@@ -219,48 +219,48 @@ members = [
 #
 # This list is ordered alphabetically.
 [profile.dev.package]
-blake2 = { opt-level = 3 }
-blake2-rfc = { opt-level = 3 }
-blake2b_simd = { opt-level = 3 }
-chacha20poly1305 = { opt-level = 3 }
+blake2            = { opt-level = 3 }
```
Comment (thiolliere) [inline Cargo.toml:222]: personally I don't like aligning, it makes every line long when only one crate has a big name.

**Comment by gilescope** [comment]
Yeah the vertical alignment is definitely marmite - you either love it or
hate it. I can see it being a PITA if I am using vi.
Let me turn that off when I get near a computer.

If we’re going to add in one more tool into the pipeline I really want it
to be clippy rather than a toml formatter. Is there value in occasionally
formatting the tomls on an ad-hoc basis? I suspect it will be a few years
before cargo fmt does it.

**Comment by gilescope** [comment]
Maybe it's too early for this.

**Pull request closed without merging**